ocaml/xapi/sm_exec.ml

```diff
@@ -117,7 +117,11 @@ let make_call ?driver_params ?sr_sm_config ?vdi_sm_config ?vdi_type
           Option.map (fun self -> Db.VDI.get_location ~__context ~self) vdi_ref
       in
       let vdi_uuid =
-        Option.map (fun self -> Db.VDI.get_uuid ~__context ~self) vdi_ref
+        match vdi_uuid with
+        | None ->
+            Option.map (fun self -> Db.VDI.get_uuid ~__context ~self) vdi_ref
```

Comment:
Shouldn't the VDI's uuid in the database match with what is being passed here? If we're forcing a specific VDI can we also update the VDI's UUID in the DB to match to avoid inconsistencies?

Reply:
My intention was that `vdi_uuid = Some uuid` is only used when creating a new VDI which is not yet in the database. I do agree that the problem is most likely in this code.

Reply:
So at the moment, the uuid of a vdi is created by the SM backend? which is not aware of the special case?